adventure-crew-game/Assets/Scripts/Backend/IAP/IAPProduct.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using UnityEngine;
+
+namespace Backend.IAP
+{
+    [Serializable]
+    public class IAPProduct
+    {
+        [SerializeField] private string id;
+        [Space, SerializeField] private string name;
+        [SerializeField, Multiline] private string description;
+        [Space, SerializeField] private string price;
+
+        public string Id => id;
+        public string Name => name;
+        public string Description => description;
+        public string Price => price;
+        public bool Purchased { get; private set; }
+
+        public void Init()
+        {
+            Purchased = PlayerPrefs.GetInt(id, 0) == 1;
+        }
+
+        internal void OnPurchaseSuccessful()
+        {
+            // Purchase logic
+            Purchased = true;
+            PlayerPrefs.SetInt(id, 1);
```

Comment:
We should evaluate how this would work during playtesting. Players coming to our game would find other players had already purchased everything and wouldn't be able to repurchase stuff. Also, what about products that can be purchased many times?

Reply:
We can put a clear player prefs call on a secret key